### PR TITLE
Fix false alarm for WRITEONLY_PROPERTY_NOT_ALLOWED_IN_RESPONSE

### PR DESCRIPTION
**Pull request by ankhyk**
to ignore the false alarm
https://portal.azure-devex-tools.com/amekpis/correctness/detail?errorId=70cc8080-d828-4cdb-87cd-1376c63c5074

### CODE REVIEW

File: lib/swaggerValidator/ajvSchemaValidator.ts [hunk 1]
```diff
@@ -215,6 +215,16 @@ const shouldSkipError = (error: ErrorObject, cxt: SchemaValidateContext) => {
     return true;
   }
 
+  // If a response has property which x-ms-secret value is true in post we can skip this error
+  if (
```
Comment (raych1) [inline lib/swaggerValidator/ajvSchemaValidator.ts:219]: Can you check if traffic validation will fail when the schema only has "x-ms-mutability": ["create","update"]?

**Review by raych1 — COMMENTED** [review]
Can you add more tests for x-ms-mutability?

File: test/liveValidatorTests.ts [hunk 2]
```diff
@@ -744,19 +744,84 @@ describe("Live Validator", () => {
...
+      const liveValidator = new LiveValidator(options);
+      await liveValidator.initialize();
+      const payload = require(`${__dirname}/liveValidation/payloads/xmsSecretAndPOST/xmsSecretAndPOST_2021-03-01.json`);
+      const result = await liveValidator.validateLiveRequestResponse(payload);
+      assert.equal(result.responseValidationResult.isSuccessful, false);
+    });
+
+    it(`should report error in response for httpMethod which is not POST,
+    when x-ms-secret is "true" or x-ms-mutability is "create" and "update"`, async () => {
+      const swaggers = [
```
Comment (raych1) [inline test/liveValidatorTests.ts]: What's the difference between these three swaggers? can you add the main difference in the comment for better understanding?

Comment (ankhyk) [reply test/liveValidatorTests.ts]: I've added comments in test
The general structure is as follows

- [√]httpMethod: POST, [√]x-ms-secret: "true", [X]x-ms-mutability: "create" and "update"
- [√]httpMethod: POST, [√]x-ms-secret: "true", [√]x-ms-mutability: "create" and "update"
- [√]httpMethod: POST, [X]x-ms-secret: "true", [√]x-ms-mutability: "create" and "update"
- [X]httpMethod: POST
  - [√]x-ms-secret: "true", [X]x-ms-mutability: "create" and "update"
  - [√]x-ms-secret: "true", [√]x-ms-mutability: "create" and "update"
  - [X]x-ms-secret: "true", [√]x-ms-mutability: "create" and "update"

File: test/liveValidatorTests.ts [hunk 3]
```diff
@@ -738,88 +738,97 @@ describe("Live Validator", () => {
...
+            git: {
+              shouldClone: false,
+            },
+          };
+          const liveValidator = new LiveValidator(options);
+          await liveValidator.initialize();
+          const payloadVersion = swagger.slice(0, -7);
+          const payload = require(`${__dirname}/liveValidation/payloads/xmsSecretAndPOST/xmsSecretButGet_${payloadVersion}.json`);
+          const result = await liveValidator.validateLiveRequestResponse(payload);
+          assert.equal(result.responseValidationResult.isSuccessful, false);
```
Comment (raych1) [inline test/liveValidatorTests.ts:838]: It needs to further validate if the error code is expected.

File: test/liveValidatorTests.ts [hunk 4]
```diff
@@ -838,8 +839,9 @@ describe("Live Validator", () => {
 					const errors = result.responseValidationResult.errors;
 					for (const error of errors) {
 						assert.equal(
-							error.code === "WRITEONLY_PROPERTY_NOT_ALLOWED_IN_RESPONSE" ||
-								error.code === "SECRET_PROPERTY",
+							(error.schemaPath.indexOf("x-ms-secret") !== -1 && error.code === "SECRET_PROPERTY") ||
```
Comment (raych1) [inline test/liveValidatorTests.ts:842]: Indent issue?

**Review by raych1 — APPROVED** [review]

**Pull request merged**